unified_planning/model/presence.py

```diff
@@ -0,0 +1,33 @@
+from unified_planning.environment import Environment
+from unified_planning.model.fnode import FNode
+from unified_planning.model.expression import (
+    NumericConstant,
+    uniform_numeric_constant,
+    TimeExpression,
+)
+from abc import ABC
+from enum import Enum, auto
+from fractions import Fraction
+from typing import Union, Optional
+
+class Presence:
+    def __init__(self, container: str):
+        self._container = container
```

Comment:
@arbimo Could the container be an Activity instead of a str here? Or you choose str for a specific reason?

Reply:
I have not given it too much thought about it and used the same pattern as for the timepoint expression (that also refers to a container), optimistically assuming I did a good choice at the time.

From what I remember the use of the string identifier was notably motivated by the avoidance of cyclic dependencies between the modules. 
Extensiblity is another one. For instance, the timepoint expression might either refer to a task (HTN) or an activity (scheduling)

I agree it is weak and a potential footgun and would be very pleased with a more type-safe solution.